daemon/exec.go

```diff
@@ -248,3 +248,34 @@ func (d *Daemon) Exec(c *Container, execConfig *execConfig, pipes *execdriver.Pi
 
 	return exitStatus, err
 }
+
+// execCommandGC starts a go routine on a ticker to clean up the daemon references
+// of exec configs that are no longer part of the container.
+func (d *Daemon) execCommandGC() {
+	for range time.Tick(5 * time.Minute) {
+		var (
+			cleaned          int
+			liveExecCommands = d.containerExecIds()
+			ids              = d.execCommands.List()
+		)
+		for _, id := range ids {
+			if _, exists := liveExecCommands[id]; !exists {
```

Comment:
@crosbymichael the problem with time-based stuff is that its inherently racey because inevitably someone will complain about it happening too fast. This puts control in the hands of the user - we just set the boundaries - 10 execs in this case.